### DISCUSSION TITLE
[Bug Report] Double quotes occurring within an element not being escaped in JSON

### DISCUSSION THREAD

**Discussion by timothylevine** [etsy/open-api#1267] [Q&A]
In the response for getShopReceipts, any double quotes that may occur within any of the resulting fields are not being escaped in the JSON that is being generated by the API.  Particular problem fields are "description" within the "transactions" array and "personalization" within the "variations" array within the "transactions" array.  The API will build invalid JSON if double quotes occur within those fields presumably amongst other fields in which double quotes could occur.

**Comment by timothylevine** [comment]
Really!?!  Is no one going to respond to an objective, repeatable bug in the API?  The API is generating malformed JSON.  That's a BIG problem.

**Comment by timothylevine** [comment]
Note that this is happening with JavaScript fetch.  A colleague of mine tells me that it does not happen in php curl but that php curl will cause several high ASCII characters to get mangled.

**Comment by Gareth064** [comment]
Etsy developers don't use this issues board to monitor for reports of bugs etc. You'll need to email them directly to ask for help

developer@etsy.com

As for what you are reporting. Are you saying that if the personalisation field said _"I would really **"love"** you to add my name to this"_ then the double quotes around love are not being escaped?

My app's main usage of the API is the getShopReciepts endpoint and I have no issues with the JSON that is returned. I am running on .NET.

**Reply by timothylevine** [reply]
Are you using php curl or JavaScript fetch?  It is happening as you describe above but only from JavaScript fetch.

**Reply by timothylevine** [reply]
I also wasn't aware that the developers didn't monitor this board since this where the API documentation directs you for feedback.

**Reply by Gareth064** [reply]
I can't tell you what .NET C# uses under the hood, but it definitely won't be JavaScript with this being a Microsoft .NET product.

**Reply by Gareth064** [reply]
Yeah when they were building v3 of the API, they were active on here working with us to test and fix bugs initially. Then they just abandoned it with no official word to say what was happening. We only know they don't monitor it, and its classed now as "community support" is because they have told some of us individually in emails when we've emailed to ask whats going on. It feels like they have got the API to a place "they" are happy with and thats as far as it goes. No more working with the community to fix further reported bugs ot to introduce community asked features etc.

**Reply by timothylevine** [reply]
More on this.  Digging deeper, I discovered that the double quotes within elements are URL Encoded as \&quot; without a backslash preceding them.  If the API response is URL decoded prior to the API response JSon being parsed it will result in malformed JSon.  A workaround would be either to first Parse the JSon then URL Decode or alter whatever you're using to URL Decode to add the escape for quotes.  More ideal would probably be for the API to include the backslashes before \&quot; so that you have correctly formed JSon after the API results are URL Decoded.
